README.md

```diff
@@ -1,54 +1,81 @@
+  * `idpTagMap`: _Optional_, a map whose `keys` identify IdP Tags which can be mapped to assurance values. IdP Metadata, of the authenticating user, contain any of the list of the allowed values, then the module will append these values into the State. Defaults to `array()`.
 
 #### Example
 This filter should be configured on IdP:
 - Specific for only one hosted IdP in `saml20-idp-hosted.php` or `shib13-idp-hosted.php`.
 ```
-    authproc = array(
-        ...
-        40 => array(
-            'class' => 'assurance:DynamicAssurance',
-            'attribute' => 'eduPersonAssurance',
-            'candidates' => array(
-                'https://refeds.org/profile/sfa',
-                'https://refeds.org/profile/mfa',
-            ),
-            'entitlementWhitelist' => array(
-                'urn:mace:www.example.org:entitlement01',
-                'urn:mace:www.example.org:entitlement02',
-            ),
-            'defaultAccurance' => 'https://example.org/LowAssurance',
-            'defaultElevatedAssurance' => 'https://example.org/HighAssurance',
-            'idpPolicies' => array(
-                'example.org:policy01',
-                'example.org:policy02',
-            ),
-            'idpTags' => array(
-                'exampleTag01',
-                'exampleTag02',
-            ),
-        ),
+     authproc = array(
+          ...
+          40 => array(
+              'class' => 'assurance:DynamicAssurance',
+              'attribute' => 'eduPersonAssurance',
+              'assuranceMap' => array(
+                  'eduPersonAssurance' => array(
+                      '1.2.840.113612.5.2.2.1' => array(                    // Classic
+                          'https://example.org/profile/Assurance/Low',
+                      ),
+                      '1.2.840.113612.5.2.2.5' => array(                    // MICS
+                          'https://example.org/profile/Assurance/High',
+                      ),
+                      '/^https:\/\/example\.org\/assurance/m',              // Pass Through values
+                  ),
+                  'eduPersonEntitlement' => array(
+                      'vo_test:IdP Proxy test' => array(
+                          'https://example.org/LoA#AssuranceHigh',
+                      ),
+                      'vo_test2:IdP Proxy test2' => array(
+                          'https://example.org/LoA#AssuranceLow',
+                      ),
+                  ),
+                  // The Attribute maps to an assurance level. All Assurance values must be underneath the key zero(0)
+                  'voPersonVerifiedEmail' => array(
+                      0 => array(
+                          'https://example.org/LoA#AssuranceLow',
+                      ),
+                  ),
```

Comment:
Can't we use a regex instead?
```
'voPersonVerifiedEmail' => array(
    '/.+$/' => array(
        'https://example.org/LoA#AssuranceLow',
    ),
),
```
So the generic attribute map syntax becomes:
```
'<attribute_name> => array(
    '<attribute_value_regex>` => array(
        'assurance_value1',
        'assurance_value2',
    ),
),
```

Reply:
Done